Build/Build.cs

```diff
@@ -65,13 +70,15 @@ class Build : NukeBuild
     string YarnCli => ToolPathResolver.GetPackageExecutable("Yarn.MSBuild", "yarn.js", "1.22.19");
 
     Target Clean => _ => _
+        .OnlyWhenDynamic(() => SourceChangesDetected())
```

Comment:
🤔 I _think_ you only need the conditional on the `Push` target, but I'm not 100% sure.

Reply:
No, unfortunately not. I had similar confusions in my day job. The `OnlyWhen(Dynamic|Static)` only applies to the `Executes` part. Dependencies are still executed and have to be skipped.